### PR TITLE
feat(LogViewer): refactor toolbar functionality

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: #5738

**In this PR**:
- Removes `LogViewerCustomControl`, `LogViewerToolbar`
- Adds `LogViewerSearch` containing search input
- Adds `toolbar` property to `LogViewer` (replacing `hasToolbar` and `customControls`)
- Adds `LogViewerToolbarContext` to wire up `LogViewerSearch` wherever it is passed within a toolbar

**Notes**:
- The `toolbar` property could possibly be named `header` instead, as this prop represents all content passed under the `pf-c-log-viewer__header`. Any preferences?
- `LogViewerSearch` will only function properly when passed into `LogViewer` via the `toolbar` prop due to the position of the context used to pass relevant data.

**Todos** (either here or in a follow-up):
- Allow persisted `searchedInput`
- Better complex toolbar example

https://patternfly-react-pr-5769.surge.sh/components/log-viewer

### CODE REVIEW

File: packages/react-log-viewer/src/LogViewer/examples/LogViewer.md [hunk 1]
```diff
@@ -2,70 +2,102 @@
...
-    </React.Fragment>
-  );  
+  return (
+    <LogViewer
+      data={data.data}
+      toolbar={
+        <Toolbar>
+          <ToolbarContent>
+            <ToolbarItem>
+              <ToolbarGroup>
```
Comment (nicolethoen) [inline packages/react-log-viewer/src/LogViewer/examples/LogViewer.md:41]: I dont think you need to wrap the SearchInput in a ToolbarGroup

File: packages/react-log-viewer/src/LogViewer/examples/LogViewer.md [hunk 2]
```diff
@@ -2,70 +2,102 @@
...
+      toolbar={
+        <Toolbar>
+          <ToolbarContent>
+            <ToolbarItem>
+              <Button onClick={onActionClick} variant="control">
+                Test Action
+              </Button>
+            </ToolbarItem>
+            <ToolbarItem>
+              <ToolbarGroup>
```
Comment (nicolethoen) [inline packages/react-log-viewer/src/LogViewer/examples/LogViewer.md:88]: same here

**Comment by kmcfaul** [comment]
Extends SearchInputProps so user can hook into the component a bit better. Props are spread before several functions to allow users to hook into change/click events without overriding internal wireup.

**Review by tlabaj — APPROVED** [review]

**Review by nicolethoen — APPROVED** [review]

**Review by nicolethoen — COMMENTED** [review]
the searches actually are not working atm. The values do not update

**Review by nicolethoen — CHANGES_REQUESTED** [review]
the searches are not working atm in the examples

**Comment by nicolethoen** [comment]
Looks good - only question is why does it automatically put focus into the second example's search?

**Comment by kmcfaul** [comment]
Do you mean the automatic scroll to the first match? It did that before, but I think we can adjust it if that isn't the desired behavior.

**Review by nicolethoen — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]

**Pull request merged**